### ISSUE TITLE
[BUG] No graphical interface on Arch Linux

### ISSUE THREAD

**Issue by popsUlfr** [Dummerle/Rare#141]
Hello!

**Describe the bug**

Launching rare does not produce any graphical interface (only the tray icon).
Applies to AUR install, pip or AppImage.
The dialog boxes during setup appear though and I could link my account.

**To Reproduce**
1. Launch rare

**Expected behavior**
A graphical interface should appear ?

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Arch Linux
- Installation AUR/pip/AppImage
- Python version 3.10.1
- DE: KDE (X11)
- GPU: Nvidia 980Ti 470.62.16

**Additional context**

**Error message**

`rare --debug`

```
[Rare] INFO: Launching Rare version 1.8.0 Codename: Stellula Kakopo
Using Legendary 0.20.21 Codename: Anticitizen One (hotfix #2) as backend
Operating System: Linux, Python version: 3.10.1
Running /usr/bin/python /usr/bin/rare --debug
[LGDLFS] DEBUG: Loading installed games failed: FileNotFoundError(2, 'No such file or directory')
[Core] DEBUG: Set locale to en-GB
[Core] INFO: Trying to re-use existing login session...
[Login] INFO: You are logged in
[Login] DEBUG: Api Request got from gamelist
[Login] DEBUG: Api Request got from mac
[LGDLFS] DEBUG: Generating list of aliases...
[Login] DEBUG: Api Request got from no_assets
[Login] DEBUG: Api Request got from assets
[Login] DEBUG: Api Request got from 32bit
[Login] DEBUG: Api Request got from saves
[Login] INFO: App starting
[LGDLFS] DEBUG: Failed to load installed game data: FileNotFoundError(2, 'No such file or directory')
[LegendarySettings] DEBUG: Set config install_dir to /home/popsulfr/legendary
[Ubisoft] ERROR: No linked ubisoft account found! Link your accounts via your browser and try again.
```

**Comment by loathingKernel** [comment]
What window manager?

**Comment by popsUlfr** [comment]
> What window manager?

KDE (kwin) on X11.

**Comment by Dummerle** [comment]
Does double click or right click on the tray icon work?

**Comment by popsUlfr** [comment]
> Does double click or right click on the tray icon work?

Double click or click does nothing.
Right click shows the menu with:
* Rare
* Exit

Clicking on Rare does nothing.

**Comment by MultisampledNight** [comment]
Partly able to reproduce when running under KDE. Rare opens up fine on the start, but if the option `Exit to System tray` is on and the main window closes, it is indeed impossible to open it again by only left-clicking, but right-clicking and selecting `Rare` successfully opens Rare again.
But I found something else. On a floating window manager such as KWin, you can easily move Rare out of the accessible screen by accident, for example by moving it under the taskbar:
![image](https://user-images.githubusercontent.com/80128916/146642711-bed9a557-f8da-4935-ac5f-5393a53a8b7f.png)
Which might also happen if you hit an unfortunate hotkey, causing Rare to get moved out of the screen completely. Opening Rare seems to do nothing then.

TLDR: Maybe Rare is opened already, try moving it to the visible screen-space with Super + Arrow Left.

**Comment by popsUlfr** [comment]
I tried window rules and they get applied to the initial "Launching Rare" dialog but the main gui is nowhere to be seen :(

I only see this dialog when I start Rare and then nothing:

![image](https://user-images.githubusercontent.com/1030423/146643396-17208be9-b3e0-4fa8-8265-7c74fe1f9fad.png)

**Comment by popsUlfr** [comment]
Ok, if I comment `self.hide()` in rare/components/main_window.py I see the window now.

```diff
diff --git a/rare/components/main_window.py b/rare/components/main_window.py
index 1eae578..583c948 100644
--- a/rare/components/main_window.py
+++ b/rare/components/main_window.py
@@ -41,7 +41,7 @@ class MainWindow(QMainWindow):
         # get the margins of the decorated window
         margins = self.windowHandle().frameMargins()
         # hide the window again because we don't want to show it at this point
-        self.hide()
+        #self.hide()
         # get the screen the cursor is on
         current_screen = QApplication.screenAt(QCursor.pos())
         # get the available screen geometry (excludes panels/docks)
```

**Comment by loathingKernel** [comment]
That is not the solution. The window is not supposed to be shown there, but when the loading dialog has finished. The reason the window is shown at that point is to be decorated. This looks more like a user configuration or KDE peculiarity thus far according to what @MultisampledNight commented.

**Comment by popsUlfr** [comment]
I see that `self.show()` is only called in `timer_finished` but that part is never reached when executing rare on my end.

```python
    def timer_finished(self):
        file_path = os.path.join(data_dir, "lockfile")
        if os.path.exists(file_path):
```

It never reaches the condition that the lockfile exists is that normal ? The timer loops forever.

> Can you give us some more information about your desktop environment?

What do you want to know ? It's KDE with no advanced changes.

**Comment by Dummerle** [comment]
Maybe I found a solution:

```diff
diff --git a/rare/components/main_window.py b/rare/components/main_window.py
index 1eae578..f3eb16f 100644
--- a/rare/components/main_window.py
+++ b/rare/components/main_window.py
@@ -38,10 +38,9 @@ class MainWindow(QMainWindow):
         self.move(-50000, -50000)
         # show the window in order for it to get decorated, otherwise windowHandle() is null
         self.show()
+
         # get the margins of the decorated window
         margins = self.windowHandle().frameMargins()
-        # hide the window again because we don't want to show it at this point
-        self.hide()
         # get the screen the cursor is on
         current_screen = QApplication.screenAt(QCursor.pos())
         # get the available screen geometry (excludes panels/docks)
@@ -49,6 +48,10 @@ class MainWindow(QMainWindow):
         decor_width = margins.left() + margins.right()
         decor_height = margins.top() + margins.bottom()
         window_size = QSize(width, height).boundedTo(screen_rect.size() - QSize(decor_width, decor_height))
+
+        # hide the window again because we don't want to show it at this point
+        self.hide()
+
         self.resize(window_size)
         self.move(screen_rect.center() - self.rect().adjusted(0, 0, decor_width, decor_height).center())
 

```

timer_finished() has nothing to do with it. You can test, if you want

**Comment by popsUlfr** [comment]
@Dummerle Thanks for the suggestion! Unfortunately it does not fix the problem.

**Comment by loathingKernel** [comment]
The reason that `self.show()` shouldn't be in there is because we don't want to show the window in case the user launched a game directly instead of Rare. If it is in the constructor, it will be shown unconditionally and it will flicker before being finally hidden.

**Comment by Dummerle** [comment]
On my clean KDE test vm it worked. Try to move `self.hide()` behind `self.move(screen...)`.

**Comment by popsUlfr** [comment]
> The reason that `self.show()` shouldn't be in there is because we don't want to show the window in case the user launched a game directly instead of Rare. If it is in the constructor, it will be shown unconditionally and it will flicker before being finally hidden.

Yeah, that makes sense. But it's weird, it's like if there's no last `self.show()` in the `__init__` the window just never comes back.

> On my clean KDE test vm it worked. Try to move `self.hide()` behind `self.move(screen...)`.

Didn't change the problem.

I tried launching with `--silent` and I'm getting a core dump I'm not sure if it could be related:

```
[Rare] INFO: Launching Rare version 1.8.2 Codename: Stellula Kakopo
Using Legendary 0.20.20 Codename: Anticitizen One (hotfix) as backend
Operating System: Linux, Python version: 3.10.1
Running /usr/bin/python3 /home/popsulfr/work/Rare/rare/__main__.py --debug --silent
[LGDLFS] DEBUG: Loading installed games failed: FileNotFoundError(2, 'No such file or directory')
[Core] DEBUG: Set locale to en-GB
[Core] INFO: Trying to re-use existing login session...
[Login] INFO: You are logged in
[Login] DEBUG: Api Request got from gamelist
[Login] DEBUG: Api Request got from mac
[LGDLFS] DEBUG: Generating list of aliases...
[Login] DEBUG: Api Request got from no_assets
[Login] DEBUG: Api Request got from 32bit
[Login] DEBUG: Api Request got from assets
[Login] DEBUG: Api Request got from saves
[Login] INFO: App starting
[LGDLFS] DEBUG: Failed to load installed game data: FileNotFoundError(2, 'No such file or directory')
[LGDLFS] DEBUG: Failed to load installed game data: FileNotFoundError(2, 'No such file or directory')
[LegendarySettings] DEBUG: Set config install_dir to /home/popsulfr/legendary
[Ubisoft] ERROR: wrapped C/C++ object of type Signals has been deleted
Error
[Rare] CRITICAL: Traceback (most recent call last):
  File "/home/popsulfr/work/Rare/rare/components/tabs/settings/ubisoft_activation.py", line 35, in run
    self.signals.worker_finished.emit(set(), set(), "")
RuntimeError: wrapped C/C++ object of type Signals has been deleted

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/popsulfr/work/Rare/rare/components/tabs/settings/ubisoft_activation.py", line 47, in run
    self.signals.worker_finished.emit(set(), set(), "error")
RuntimeError: wrapped C/C++ object of type Signals has been deleted

QWidget: Must construct a QApplication before a QWidget
[1]    1047613 abort (core dumped)  python3 -m rare --debug --silent
```

**Comment by loathingKernel** [comment]
This is a weird interaction and I do not think it has to do with the window itself. Can you try one more thing?

`-         self.move(-50000, -50000)`
`+         self.move(-2000, -2000)`

Just in case KDE has an issue with the window being moved to far outside the viewport.

**Comment by popsUlfr** [comment]
> This is a weird interaction and I do not think it has to do with the window itself. Can you try one more thing?

Yes, of course thank you for all the help!

It doesn't change anything.

**Comment by Dummerle** [comment]
@popsUlfr I fixed the silent mode, you can try it, if you run it from source. Maybe I fixed the issue, that no window appears, but I can't reproduce it.

**Comment by popsUlfr** [comment]
> @popsUlfr I fixed the silent mode, you can try it, if you run it from source. Maybe I fixed the issue, that no window appears, but I can't reproduce it.

@Dummerle YES! It's fixed now! \o/ Thank you. Silent mode and normal mode work and the window appears properly.
